### PR TITLE
removed OrderedRDD

### PR DESCRIPTION
**Pull request by cseed**
This isn't quite ready, but PR'ing to draw a line in the sand.

There are compile errors in concordance, nirvana, vep and MT.  They should all be fixed by the pending PRs with the exception of a few additional fixups in MT (fromLegacy, same, etc.)  I will rebase this as those PRs go in and fix up anything else that remains.

### CODE REVIEW

**Comment by cseed** [comment]
Now depends on https://github.com/hail-is/hail/pull/2723, last pending rv-conversion PR.

 > +275 −2,223

Aw, yiss.  @tpoterba Do you want to do the honors on this or should I spin the wheel?

**Comment by cseed** [comment]
I spun the wheel and @catoverdrive was the winner.

File: src/main/scala/is/hail/variant/MatrixTable.scala [hunk 1]
```diff
@@ -1749,12 +1736,18 @@ class MatrixTable(val hc: HailContext, val metadata: VSMMetadata,
...
+    val localRowType = rowType
 
     Table(hc,
-      rdd.mapPartitions { it =>
+      rdd2.mapPartitions { it =>
         val n = vNames.length + gNames.length * localNSamples
 
-        it.map { case (v, (va, gs)) =>
+        it.map { rv =>
+          val ur = new UnsafeRow(localRowType, rv)
```
Comment (catoverdrive) [inline src/main/scala/is/hail/variant/MatrixTable.scala]: I think you can move the `UnsafeRow` constructor outside of the `it.map` if you want, here.

File: src/main/scala/is/hail/variant/MatrixTable.scala [hunk 2]
```diff
@@ -1819,11 +1812,17 @@ class MatrixTable(val hc: HailContext, val metadata: VSMMetadata,
...
     val localSampleAnnotationsBc = sampleAnnotationsBc
+    val localRowType = rowType
 
-    val result = rdd.mapPartitions { it =>
+    val result = rdd2.mapPartitions { it =>
       val zv = zVal.map(_.copy())
       ec.set(0, globalBc.value)
-      it.foreach { case (v, (va, gs)) =>
+      it.foreach { rv =>
+        val ur = new UnsafeRow(localRowType, rv)
```
Comment (catoverdrive) [inline src/main/scala/is/hail/variant/MatrixTable.scala]: same here.

File: src/main/scala/is/hail/variant/MatrixTable.scala [hunk 3]
```diff
@@ -2081,12 +2080,30 @@ class MatrixTable(val hc: HailContext, val metadata: VSMMetadata,
...
+          that.rdd2.rdd)
+          .rdd) { (it1, it2) =>
         var partSame = true
         while (it1.hasNext && it2.hasNext) {
-          val (v1, (va1, gs1)) = it1.next()
-          val (v2, (va2, gs2)) = it2.next()
+          val rv1 = it1.next()
+          val rv2 = it2.next()
+
+          val ur1 = new UnsafeRow(rowType1, rv1)
```
Comment (catoverdrive) [inline src/main/scala/is/hail/variant/MatrixTable.scala]: and here.

File: src/main/scala/is/hail/methods/Nirvana.scala [hunk 4]
```diff
@@ -285,17 +285,17 @@ object Nirvana {
...
+    val annotations = vds.rdd2
+      .mapPartitions { it =>
         val pb = new ProcessBuilder(cmd.asJava)
         val env = pb.environment()
         if (path.orNull != null)
           env.put("PATH", path.get)
 
-        it.filter { case (v, va) =>
-          rootQuery.forall(q => q(va) == null)
+        it.map { rv =>
```
[the remark this answers is not in the document]
Comment (cseed) [reply src/main/scala/is/hail/methods/Nirvana.scala]: Yeah, so we had this strategy (in practice the added complexity wasn't worth it and it was never really used) to precompute VEP on common variants and join against the precomputed values and then run VEP on the remaining variants that were un-annotated (missing).  Nirvana (which is much faster and this strategy is even less useful) picked up this code from VEP.  The code was already removed from VEP, I just did the same thing for Nirvana.

I admit, it was sneaky to include in this PR.

**Pull request merged**